dateparser/languages/language.py

```diff
@@ -1,29 +1,46 @@
 # -*- coding: utf-8 -*-
+import hashlib
 import re
 from itertools import chain
 
 from dateutil import parser
 
 from dateparser.timezone_parser import pop_tz_offset_from_string
-from dateparser.utils import wrap_replacement_for_regex
+from dateparser.utils import wrap_replacement_for_regex, registry
 
 from .dictionary import Dictionary, ALWAYS_KEEP_TOKENS
 from .validation import LanguageValidator
 
 
+@registry
 class Language(object):
+
     _dictionary = None
     _splitters = None
     _wordchars = None
 
-    def __init__(self, shortname, language_info):
+    @classmethod
+    def get_key(cls, *args, **kwargs):
+        key_template = '%(shortname)s-%(skip_tokens)s'
```

Comment:
As you suggested, with new changes, `Language` class isn't a registry - `translate`/ `is_applicable` are called with `settings` instance, which we attach to `Dictionary` instance. Now we cache `sorted_words` (a list of words) and `split_regex` (complied regexes) only in `Dictionary` class for subsequent reuse.